spec/bolt/inventory/inventory2_spec.rb

```diff
@@ -682,6 +687,329 @@ def common_data(transport)
         end
       end
     end
+
+    describe 'get_target' do
+      context 'empty inventory' do
+        let(:inventory) { Bolt::Inventory::Inventory2.new({}, config) }
+
+        it 'should parse a single target URI' do
+          name = 'notarget'
+          target = inventory.get_target(name)
+          expect(target.name).to eq(name)
+        end
+
+        it 'should error when an array of targets is requested' do
+          names = ['pcp://a', 'winrm://b', 'c']
+          expect {
+            inventory.get_target(names)
+          }.to raise_error(Bolt::Inventory::ValidationError, "'#{names}' refers to 3 targets")
+        end
+
+        it 'should fail for unknown protocols' do
+          expect {
+            inventory.get_target('z://foo')
+          }.to raise_error(Bolt::UnknownTransportError, %r{Unknown transport z found for z://foo})
```

Comment:
Should this also verify an error is raised with an invalid URI?

Reply:
We have a test in group2 to test that invalid URI's are forbidden. Were you suggesting a new test here with an invalid uri? (`z://foo`) is valid.

Reply:
I was suggesting adding a new test here, either directly testing `parse_uri` or exercising the `parse_uri` call in `inventory2` in addition to `group2`. I don't think it's worth holding up this PR for though.